numpy/_core/src/multiarray/number.c

```diff
@@ -60,24 +60,24 @@ array_inplace_matrix_multiply(PyArrayObject *m1, PyObject *m2);
  * Those not present will not be changed
  */
 
-/* FIXME - macro contains a return */
-#define SET(op)   temp = _PyDict_GetItemStringWithError(dict, #op); \
-    if (temp == NULL && PyErr_Occurred()) { \
+/* FIXME - macro contains returns  */
+#define SET(op) \
+    res = PyDict_GetItemStringRef(dict, #op, &temp); \
+    if (res == -1) { \
         return -1; \
     } \
-    else if (temp != NULL) { \
+    else if (res == 1) { \
         if (!(PyCallable_Check(temp))) { \
```

Comment:
missing: Py_DECREF(temp);

Reply:
This is a spot where a borrowed reference was upgraded to an owned reference. I deleted the `INCREF` that we had before to deal with the borrowed reference, and then used `Py_SETREF` so `n_ops.op` owns the reference after that. I'm confused why I'd need to still `DECREF` `temp`

Reply:
`PyDict_GetItemStringRef(&temp)` creates a strong reference. This line (`return -1;`) does exit the function. I don't see anything which clears the strong reference.

Reply:
ah sorry I missed you were talking about the error cases, thanks for explaining a bit more